### PR TITLE
Exemptions refactor and features

### PR DESCRIPTION
**Pull request by Bnaya**
refactor for the exemption page
implementation for:
#324  #319  #314
Group exemptions by ministry name and not branch 

Using backbone router for the exemption page startup

Live version:
http://obudget.bnaya.net/#spending//2014/main

### CODE REVIEW

**Comment by Bnaya** [comment]
This PR is not for merge yet,
We want to first do the mobile adjustments for the current code

File: open-budget/app/scripts/Hasadna/oBudget/Pages/Exemptions/dataHelpers.coffee [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        variantsArray.forEach (variantFragment) ->
+          if (ministryVariant.indexOf variantFragment) > -1
+            matched = ministry
+
+      if !matched
+        matched = ministryVariant
+
+      matched
+
+    getExemptionsExportUrl: (exemptions) ->
```
Comment (akariv) [inline open-budget/app/scripts/Hasadna/oBudget/Pages/Exemptions/dataHelpers.coffee]: why go through this trouble when you have a CSV output option from the server?

File: open-budget/app/scripts/Hasadna/oBudget/Pages/Exemptions/DataStruct/Entity.coffee [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          id: null
+          exemptions_by_publisher: {}
+          exemptions_sum: null
+
+      initialize: (attrs, options) ->
+        @baseURL = options.baseURL
+        @entityId = options.entityId
+
+      doFetch: ->
+        @fetch(success: @handleFetchResult)
```
Comment (akariv) [inline open-budget/app/scripts/Hasadna/oBudget/Pages/Exemptions/DataStruct/Entity.coffee]: why remove the datatype here?

File: open-budget/app/scripts/Hasadna/oBudget/Pages/Exemptions/DataStruct/Exemption.coffee [hunk 3]
```diff
@@ -0,0 +1,20 @@
+define ['backbone', 'Hasadna/oBudget/Pages/Exemptions/DataStruct/SpendingLine'], (Backbone, SpendingLine) ->
+
+  class Exemption extends Backbone.Model
+
+    initialize: (attrs, options) ->
+      @baseURL = options.baseURL
+      @publicationId = options.publicationId
+
+    doFetch: ->
+      @fetch(success: @handleFetchResult)
```
Comment (akariv) [inline open-budget/app/scripts/Hasadna/oBudget/Pages/Exemptions/DataStruct/Exemption.coffee]: why remove the datatype here?

**Pull request merged**